Auth/app/src/main/java/com/example/Auth/BioActivity.kt

```diff
@@ -0,0 +1,316 @@
+                            val textViewDescr = mapData["descr"].toString()
+
+                            val knopka =
+                                adapter.addKnopka(Knopka(textViewLabel, "", 0, 0))
+
+                            val m: Map<String, String?> = mapOf(
+                                "\"name\"" to "\"" + mapData["name"] + "\"",
+                                "\"style\"" to /*"\"" + mapData["style"].toString() + "\""*/ null,
+                                "\"pushes\"" to "0",
+                                "\"id\"" to units.id.toString()
+//                                "\"LocalDateTime\"" to "\"" + mapData["LocalDateTime"] + "\""
+                            )
+
+                            sendPostButtonRequest(
+                                m as Map<String, String>,
+                                knopka,
+                                knopka.knopkaList.size - 1
+                            ) // the end of the knpokas list
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun storageInfoUpdate()/*: Boolean*/ {
+
+        val imageString: String? = bitMapToBase64String(units.profilePicBitMap)
+        val nameString: String = units.textViewName?.text.toString()
+        val bioString: String = units.textViewBio?.text.toString()
+        val tokenString: String = units.token.toString()
+        val sharedPref = getSharedPreferences("mypref", 0)
+        val editor = sharedPref.edit()
+
+
+        editor.putString("nickname", nameString)
+        editor.putString("bio", bioString)
+        editor.putString("image", imageString)
+        editor.putString("token", tokenString)
+
+        editor.apply()
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun storageInfoLoad() /*: Boolean*/ {
+
+        val sharedPref = getSharedPreferences("mypref", 0)
+
+        val nameString = sharedPref.getString("nickname", "").toString()
+        val bioString = sharedPref.getString("bio", "").toString()
+        val imageString = sharedPref.getString("image", "").toString()
+
+        if (base64StringToBitMap(imageString) != null)
+            units.profilePicBitMap = base64StringToBitMap(imageString)
+        units.imageViewProfilePic?.setImageBitmap(units.profilePicBitMap)
+        units.textViewName?.text = nameString
+        units.textViewBio?.text = bioString
+    }
+
+    private fun storageInfoClear() /*: Boolean*/ {
+        val sharedPref = getSharedPreferences("mypref", 0)
+        sharedPref.edit().clear().apply()
+    }
+
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun sendGetUserProfileInfo() {
+        val result = Requests.getUserProfileInfo().execute().get()
+        val mapData: User = jsonFormat.decodeFromString(result.toString())
+        units.textViewName?.text = mapData.nickname
+        units.textViewBio?.text = mapData.bio
+        if (mapData.photo != "") {
```

Comment:
It can be null, or empty space string, or new line ("\n" or "\r\n").
The better way to check for empty string is to use any library (like TextUtils.isEmpty())